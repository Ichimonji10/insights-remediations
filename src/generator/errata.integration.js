'use strict';

const { request } = require('../test');

test('generates a simple playbook with single RHSA remediation', async () => {
    const data = {
        issues: [{
            id: 'vulnerabilities:RHSA-2018:0502',
            systems: ['a8799a02-8be9-11e8-9eb6-529269fb1459']
        }]
    };

    const res = await request
    .post('/v1/playbook')
    .send(data)
    .expect(200);
    expect(res.text).toMatchSnapshot();
});

test('generates a simple playbook with multiple erratum-based remediation', async () => {
    const data = {
        issues: [{
            id: 'vulnerabilities:RHSA-2018:0502',
            systems: ['a8799a02-8be9-11e8-9eb6-529269fb1459']
        }, {
            id: 'vulnerabilities:RHBA-2007:0331',
            systems: ['53fbcd90-9c8f-11e8-98d0-529269fb1459']
        }]
    };

    const res = await request
    .post('/v1/playbook')
    .send(data)
    .expect(200);
    expect(res.text).toMatchSnapshot();
});

test('aggregates multiple errata into a single play', async () => {
    const data = {
        issues: [{
            id: 'vulnerabilities:RHSA-2018:0502',
            systems: ['a8799a02-8be9-11e8-9eb6-529269fb1459']
        }, {
            id: 'vulnerabilities:RHSA-2017:1852',
            systems: ['a8799a02-8be9-11e8-9eb6-529269fb1459']
        }, {
            id: 'vulnerabilities:RHSA-2017:1382',
            systems: ['a8799a02-8be9-11e8-9eb6-529269fb1459']
        }, {
            id: 'vulnerabilities:RHSA-2017:2679',
            systems: ['a8799a02-8be9-11e8-9eb6-529269fb1459']
        }]
    };

    const res = await request
    .post('/v1/playbook')
    .send(data)
    .expect(200);
    expect(res.text).toMatchSnapshot();
});
