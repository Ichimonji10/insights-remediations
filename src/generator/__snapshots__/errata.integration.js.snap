// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aggregates multiple errata into a single play 1`] = `
"---
# Upgrade packages to apply the following errata:
#   - Important: kernel-alt security and bug fix update (RHSA-2018:0502)
#   - Moderate: openldap security, bug fix, and enhancement update (RHSA-2017:1852)
#   - Important: sudo security update (RHSA-2017:1382)
#   - Important: kernel security update (RHSA-2017:2679)
# Identifier: (vulnerabilities:RHSA-2018:0502,vulnerabilities:RHSA-2017:1852,vulnerabilities:RHSA-2017:1382,vulnerabilities:RHSA-2017:2679,fix)
# Version: unknown
- name: fix RHSA-2018:0502,RHSA-2017:1852,RHSA-2017:1382,RHSA-2017:2679
  hosts: \\"a8799a02-8be9-11e8-9eb6-529269fb1459.example.com\\"
  become: true
  vars:
    changed_keyword: \\"Complete!\\"
  tasks:
    - name: upgrade package
      shell: yum update -d 2 -y --advisories=RHSA-2018:0502,RHSA-2017:1852,RHSA-2017:1382,RHSA-2017:2679
      register: yum_result
      changed_when: changed_keyword in yum_result.stdout

    - when: changed_keyword in yum_result.stdout
      name: set reboot fact
      set_fact:
        insights_needs_reboot: True

# Reboots a system if any of the preceeding plays sets the 'insights_needs_reboot' variable to true.
# The variable can be overridden to suppress this behavior.
- name: Reboot system (if applicable)
  hosts: \\"a8799a02-8be9-11e8-9eb6-529269fb1459.example.com\\"
  become: True
  gather_facts: False
  tasks:
    - when:
        - insights_needs_reboot is defined
        - insights_needs_reboot
      block:
        - name: Reboot system
          shell: sleep 2 && shutdown -r now \\"Ansible triggered reboot\\"
          async: 1
          poll: 0
          ignore_errors: true

        - name: Wait for system to boot up
          local_action:
            module: wait_for
            host: \\"{{ hostvars[inventory_hostname]['ansible_host'] | default(hostvars[inventory_hostname]['ansible_ssh_host'], true) | default(inventory_hostname, true) }}\\"
            port: \\"{{ hostvars[inventory_hostname]['ansible_port'] | default(hostvars[inventory_hostname]['ansible_ssh_port'], true) | default('22', true) }}\\"
            delay: 15
            timeout: 300
          become: false

- name: run insights
  hosts: \\"a8799a02-8be9-11e8-9eb6-529269fb1459.example.com\\"
  become: True
  gather_facts: False
  tasks:
    - name: run insights
      command: redhat-access-insights
      changed_when: false"
`;

exports[`generates a simple playbook with multiple erratum-based remediation 1`] = `
"---
# Important: kernel-alt security and bug fix update (RHSA-2018:0502)
# Identifier: (vulnerabilities:RHSA-2018:0502,fix)
# Version: unknown
- name: fix RHSA-2018:0502
  hosts: \\"a8799a02-8be9-11e8-9eb6-529269fb1459.example.com\\"
  become: true
  vars:
    changed_keyword: \\"Complete!\\"
  tasks:
    - name: upgrade package
      shell: yum update -d 2 -y --advisories=RHSA-2018:0502
      register: yum_result
      changed_when: changed_keyword in yum_result.stdout

    - when: changed_keyword in yum_result.stdout
      name: set reboot fact
      set_fact:
        insights_needs_reboot: True

# conga bug fix update (RHBA-2007:0331)
# Identifier: (vulnerabilities:RHBA-2007:0331,fix)
# Version: unknown
- name: fix RHBA-2007:0331
  hosts: \\"53fbcd90-9c8f-11e8-98d0-529269fb1459.example.com\\"
  become: true
  vars:
    changed_keyword: \\"Complete!\\"
  tasks:
    - name: upgrade package
      shell: yum update -d 2 -y --advisories=RHBA-2007:0331
      register: yum_result
      changed_when: changed_keyword in yum_result.stdout

    - when: changed_keyword in yum_result.stdout
      name: set reboot fact
      set_fact:
        insights_needs_reboot: True

# Reboots a system if any of the preceeding plays sets the 'insights_needs_reboot' variable to true.
# The variable can be overridden to suppress this behavior.
- name: Reboot system (if applicable)
  hosts: \\"53fbcd90-9c8f-11e8-98d0-529269fb1459.example.com,a8799a02-8be9-11e8-9eb6-529269fb1459.example.com\\"
  become: True
  gather_facts: False
  tasks:
    - when:
        - insights_needs_reboot is defined
        - insights_needs_reboot
      block:
        - name: Reboot system
          shell: sleep 2 && shutdown -r now \\"Ansible triggered reboot\\"
          async: 1
          poll: 0
          ignore_errors: true

        - name: Wait for system to boot up
          local_action:
            module: wait_for
            host: \\"{{ hostvars[inventory_hostname]['ansible_host'] | default(hostvars[inventory_hostname]['ansible_ssh_host'], true) | default(inventory_hostname, true) }}\\"
            port: \\"{{ hostvars[inventory_hostname]['ansible_port'] | default(hostvars[inventory_hostname]['ansible_ssh_port'], true) | default('22', true) }}\\"
            delay: 15
            timeout: 300
          become: false

- name: run insights
  hosts: \\"53fbcd90-9c8f-11e8-98d0-529269fb1459.example.com,a8799a02-8be9-11e8-9eb6-529269fb1459.example.com\\"
  become: True
  gather_facts: False
  tasks:
    - name: run insights
      command: redhat-access-insights
      changed_when: false"
`;

exports[`generates a simple playbook with single RHSA remediation 1`] = `
"---
# Important: kernel-alt security and bug fix update (RHSA-2018:0502)
# Identifier: (vulnerabilities:RHSA-2018:0502,fix)
# Version: unknown
- name: fix RHSA-2018:0502
  hosts: \\"a8799a02-8be9-11e8-9eb6-529269fb1459.example.com\\"
  become: true
  vars:
    changed_keyword: \\"Complete!\\"
  tasks:
    - name: upgrade package
      shell: yum update -d 2 -y --advisories=RHSA-2018:0502
      register: yum_result
      changed_when: changed_keyword in yum_result.stdout

    - when: changed_keyword in yum_result.stdout
      name: set reboot fact
      set_fact:
        insights_needs_reboot: True

# Reboots a system if any of the preceeding plays sets the 'insights_needs_reboot' variable to true.
# The variable can be overridden to suppress this behavior.
- name: Reboot system (if applicable)
  hosts: \\"a8799a02-8be9-11e8-9eb6-529269fb1459.example.com\\"
  become: True
  gather_facts: False
  tasks:
    - when:
        - insights_needs_reboot is defined
        - insights_needs_reboot
      block:
        - name: Reboot system
          shell: sleep 2 && shutdown -r now \\"Ansible triggered reboot\\"
          async: 1
          poll: 0
          ignore_errors: true

        - name: Wait for system to boot up
          local_action:
            module: wait_for
            host: \\"{{ hostvars[inventory_hostname]['ansible_host'] | default(hostvars[inventory_hostname]['ansible_ssh_host'], true) | default(inventory_hostname, true) }}\\"
            port: \\"{{ hostvars[inventory_hostname]['ansible_port'] | default(hostvars[inventory_hostname]['ansible_ssh_port'], true) | default('22', true) }}\\"
            delay: 15
            timeout: 300
          become: false

- name: run insights
  hosts: \\"a8799a02-8be9-11e8-9eb6-529269fb1459.example.com\\"
  become: True
  gather_facts: False
  tasks:
    - name: run insights
      command: redhat-access-insights
      changed_when: false"
`;
